src/Exception/RequestException.php

```diff
@@ -79,25 +79,86 @@ public static function create(
 
         $level = floor($response->getStatusCode() / 100);
         if ($level == '4') {
-            $label = 'Client error response';
+            $label = 'Client Error';
             $className = __NAMESPACE__ . '\\ClientException';
         } elseif ($level == '5') {
-            $label = 'Server error response';
+            $label = 'Server Error';
             $className = __NAMESPACE__ . '\\ServerException';
         } else {
-            $label = 'Unsuccessful response';
+            $label = 'Unsuccessful Request';
             $className = __CLASS__;
         }
 
-        $message = $label . ' [url] ' . $request->getUri()
-            . ' [http method] ' . $request->getMethod()
-            . ' [status code] ' . $response->getStatusCode()
-            . ' [reason phrase] ' . $response->getReasonPhrase();
+        // Server Error: `GET /` resulted in a `404 Not Found` response:
+        // <html> ... (truncated)
+        $message = sprintf(
+            '%s: `%s` resulted in a `%s` response',
+            $label,
+            $request->getMethod() . ' ' . $request->getUri(),
+            $response->getStatusCode() . ' ' . $response->getReasonPhrase()
+        );
+
+        $summary = static::getResponseBodySummary($response);
+
+        if (is_string($summary)) {
+            $message .= ":\n{$summary}\n";
+        }
 
         return new $className($message, $request, $response, $previous, $ctx);
     }
 
     /**
+     * Determine if a response is printable
+     *
+     * @param ResponseInterface $response
+     *
+     * @return bool
+     */
+    protected static function isResponsePrintable(ResponseInterface $response)
+    {
+        $types = [
+            // Plain text, javascript, css, etc
+            'text/.*',
+            // JSON of various types
+            'application/.*json',
+            // XML of various types, including XHTML
+            'application/.*xml',
```

Comment:
I think this is too limiting because unfortunately, some web services use funky content-types or don't even provide them sometimes (e.g., AWS). I also think maintaining a list of content-types here could lead to bloat over time (e.g., someone saying please include CSV, text, html, etc...)

I think we can do a regex on the first N characters to ensure that they are valid. Something like this could work (untested): 

```
$chunk = $response->read(80);
return preg_match('/^[\r\n\t[:print:]]*$/', $response) ? $chunk : null;
```